ambry-quota/src/main/java/com/github/ambry/quota/MaxThrottlePolicy.java

```diff
@@ -24,18 +25,51 @@
  * groups the quota usage for all the quotas.
  */
 public class MaxThrottlePolicy implements ThrottlePolicy {
+  static final long DEFAULT_RETRY_AFTER_MS = ThrottlingRecommendation.NO_RETRY_AFTER_MS;
+  static final int DEFAULT_RECOMMENDED_HTTP_STATUS = HttpResponseStatus.OK.code();
+  static final QuotaUsageLevel DEFAULT_QUOTA_WARNING_LEVEL = QuotaUsageLevel.HEALTHY;
+
+  // Percentage usage at or below this limit is healthy.
+  static final int HEALTHY_USAGE_LEVEL_LIMIT = 80;
+  // Percentage usage level at or below this limit and above healthy limit will generate warning.
+  static final int WARNING_USAGE_LEVEL_LIMIT = 95;
+  // Percentage usage level at or below this limit and above warning limit will generate critical warning. Anything above will be fatal.
+  static final int CRITICAL_USAGE_LEVEL_LIMIT = 100;
+
   @Override
   public ThrottlingRecommendation recommend(List<QuotaRecommendation> quotaRecommendations) {
     boolean shouldThrottle = false;
     Map<QuotaName, Float> quotaUsagePercentage = new HashMap<>();
-    int recommendedHttpStatus = 200;
-    long retryAfterMs = -1;
+    int recommendedHttpStatus = DEFAULT_RECOMMENDED_HTTP_STATUS;
+    long retryAfterMs = DEFAULT_RETRY_AFTER_MS;
+    QuotaUsageLevel quotaWarningLevel = DEFAULT_QUOTA_WARNING_LEVEL;
     for (QuotaRecommendation recommendation : quotaRecommendations) {
       shouldThrottle = shouldThrottle | recommendation.shouldThrottle();
       quotaUsagePercentage.put(recommendation.getQuotaName(), recommendation.getQuotaUsagePercentage());
       recommendedHttpStatus = Math.max(recommendation.getRecommendedHttpStatus(), recommendedHttpStatus);
       retryAfterMs = Math.max(recommendation.getRetryAfterMs(), retryAfterMs);
+      quotaWarningLevel = QuotaUsageLevel.fromInt(Math.max(quotaWarningLevel.ordinal(),
+          computeWarningLevel(recommendation.getQuotaUsagePercentage()).ordinal()));
+    }
```

Comment:
I don't think we need to convert the QuotaLevel back and forth. maybe just do this after for loop
```
computeWarningLevel(quotaUsagePercentage.values().stream().max())
```